packages/dara-core/dara/core/interactivity/derived_variable.py

```diff
@@ -207,7 +208,9 @@ def _restore_pydantic_models(func: Callable[..., Any], *args):
                 if typ is not None and isclass(typ) and issubclass(typ, BaseModel) and arg is not None:
                     parsed_args.append(typ(**arg))
                     continue
-                parsed_args.append(arg)
+                elif typ in encoder_registry:
+                    parsed_args.append(encoder_registry[typ]['deserialize'](arg))
+                    continue
```

Comment:
I think this loop is missing the `parsed_args.append(arg)` statement, if neither condition is satisfied we need to return the arg as-is